cmd/cosign/cli/tree.go

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/spf13/cobra"
+
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
+)
+
+func Tree() *cobra.Command {
+	c := &options.TreeOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "tree",
+		Short:   "Display supply chain security related artifacts for an image such as signatures, SBOMs and attestations",
+		Example: "  cosign tree <IMAGE>",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return TreeCmd(cmd.Context(), c.Registry, args[0])
+		},
+	}
+
+	c.AddFlags(cmd)
+	return cmd
+}
+
+const (
+	SignatureTagSuffix   = ".sig"
+	SBOMTagSuffix        = ".sbom"
+	AttestationTagSuffix = ".att"
+)
+
+func TreeCmd(ctx context.Context, regOpts options.RegistryOptions, imageRef string) error {
+	var scsaRefs []name.Tag
+	ref, err := name.ParseReference(imageRef)
+	if err != nil {
+		return err
+	}
+
+	remoteOpts := regOpts.GetRegistryClientOpts(ctx)
+
+	fmt.Fprintf(os.Stdout, "📦 Supply Chain Security Related artifacts for an image: %s\n", ref.String())
+
+	sigRef, err := ociremote.SignatureTag(ref, ociremote.WithRemoteOptions(remoteOpts...))
+	if err != nil {
+		return err
+	}
+	_, err = remote.Get(sigRef, remoteOpts...)
+	if err == nil {
+		scsaRefs = append(scsaRefs, sigRef)
+	}
+
+	attRef, err := ociremote.AttestationTag(ref, ociremote.WithRemoteOptions(remoteOpts...))
+	if err != nil {
+		return err
+	}
+	_, err = remote.Get(attRef, remoteOpts...)
+	if err == nil {
+		scsaRefs = append(scsaRefs, attRef)
+	}
+
+	sbomRef, err := ociremote.SBOMTag(ref, ociremote.WithRemoteOptions(remoteOpts...))
+	if err != nil {
+		return err
+	}
+	_, err = remote.Get(sbomRef, remoteOpts...)
+	if err == nil {
+		scsaRefs = append(scsaRefs, sbomRef)
+	}
+
+	if len(scsaRefs) == 0 {
+		fmt.Fprintf(os.Stdout, "No SCSA artifacts found for image %s\n", ref.String())
```

Comment:
Perhaps it is better to avoid abbreviations here. Just prompt Supply Chain Security Related Artifacts as above.

Reply:
Totally agree with you, it's an auto-completion issue. I've updated it according to your feedback, thanks.